### PR TITLE
Muon Analysis 2 - Fix load current run

### LINKED ISSUE #26216: Muon Analysis 2 - load current run
The load current run in the new GUI does not always get the correct run

Compare with the old GUI

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This PR fixes the `Load current run` button in Muon Analysis 2. It should now find the newest run or give an error if it has not been found.

**To test:**
Open Muon Analysis 2 GUI
Click load current run
It should load a file called `\\<Instrument>\data\<Instrument>auto_C.tmp` (the letter will vary depending on most recent run)
To check this is right open the old Muon Analysis GUI and load current run, it should be the same 

Fixes #26216

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/Muon/utilities/muon_file_utils_test.py [hunk 1]
```diff
@@ -62,6 +63,18 @@ def test_that_get_current_run_filename_throws_if_autosave_file_not_found(self):
         with self.assertRaises(ValueError):
             utils.get_current_run_filename("EMU")
 
+    def test_that_get_current_run_returns_correct_run(self):
+        utils.check_file_exists = mock.Mock(return_value=True)
+        file_name = os.sep + os.sep + "EMU" + os.sep + "data" + os.sep + "autoA"
+        file = StringIO(u"autoA")
```
Comment (AnthonyLim23) [inline scripts/test/Muon/utilities/muon_file_utils_test.py]: could we have a different name for the variable as file is used for IO

**Comment by AnthonyLim23** [comment]
When it cannot find the run, instead of throwing an error and carrying I would like a message box telling the user it failed. See https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/message_box.py

**Review by AnthonyLim23 — CHANGES_REQUESTED** [review]
see above comment

**Comment by alicerussell1** [comment]
Currently it does give a message box when the error is raised and then stops (which looks the same as if I use the message box linked), do you want it to display it as a warning in a message box and then continue by loading `auto_A.tmp` as default as it was doing?

**Comment by AnthonyLim23** [comment]
when I forced it fail, I never got a warning box as it just loaded the default of A. It would be good to tell the user that is what we are doing so a warning box would be good.

**Review by AnthonyLim23 — APPROVED** [review]

**Pull request merged**